src/transformers/trainer.py

```diff
@@ -1322,14 +1334,17 @@ def predict(self, test_dataset: Dataset, ignore_keys: Optional[List[str]] = None
 
         test_dataloader = self.get_test_dataloader(test_dataset)
 
-        return self.prediction_loop(test_dataloader, description="Prediction", ignore_keys=ignore_keys)
+        return self.prediction_loop(
+            test_dataloader, description="Prediction", ignore_keys=ignore_keys, metric_key_prefix=metric_key_prefix
+        )
 
     def prediction_loop(
         self,
         dataloader: DataLoader,
         description: str,
         prediction_loss_only: Optional[bool] = None,
         ignore_keys: Optional[List[str]] = None,
+        metric_key_prefix: Optional[str] = "eval",
```

Comment:
`metric_key_prefix` cannot be `None` so I think we should delete the `Optional` in the type hint